client/test/build.gradle

```diff
@@ -27,7 +26,7 @@ targetCompatibility = JavaVersion.VERSION_1_7
 sourceCompatibility = JavaVersion.VERSION_1_7
 
 dependencies {
-  compile "org.apache.httpcomponents:httpcore:${versions.httpcore}"
+  compile "org.elasticsearch.client:elasticsearch-rest-client:${version}"
```

Comment:
well, we really cant use ordinary ones now. Since all of the dependencies in rest client are shaded to use internal.org.apache classes, if we leak anything like `Header` into an API, we need to use the internal.org.apache since those are in the method signatures. So, we have to use our internal ones always, which is why we removed the httpcore stuff, because client/test now uses `internal.o.a.h.Header`, for example.